// SPDX-FileCopyrightText: 2025 Carnegie Mellon University - Living Edge Lab
//
// SPDX-License-Identifier: GPL-2.0-only

syntax = "proto3";
package protocol.swarm_control;

import "google/protobuf/any.proto";

/*
 * Directive for remote control from the Swarm Controller
 */
message SwarmControlRequest {
  // Since command sequencing is not built-in to ZeroMQ,
  // it must be done manually
  uint32 sequence_number = 1;
  google.protobuf.Any control_request = 2;
}

message SwarmControlResponse {
  // Since command sequencing is not built-in to ZeroMQ,
  // it must be done manually
  uint32 sequence_number = 1;
  google.protobuf.Any control_response = 2;
}
